public/components/custom_panels/helpers/utils.tsx

```diff
@@ -311,44 +329,107 @@ const createCatalogVisualizationMetaData = (
+  const startEpochTime = convertDateTimeToEpoch(startTime);
+  const endEpochTime = convertDateTimeToEpoch(endTime, false);
+  const promQuery =
+    attributesGroupBy.length === 0
+      ? catalogTableName
+      : `${aggregation} by(${attributesGroupString}) (${catalogTableName})`;
+
+  return `source = ${catalogSourceName}.query_range('${promQuery}', ${startEpochTime}, ${endEpochTime}, '${spanParam}')`;
+};
+
 // Creates a catalogVisualization for a runtime catalog based PPL query and runs getQueryResponse
-export const renderCatalogVisualization = async (
-  http: CoreStart['http'],
-  pplService: PPLService,
-  catalogSource: string,
-  startTime: string,
-  endTime: string,
-  filterQuery: string,
-  spanParam: string | undefined,
-  setVisualizationTitle: React.Dispatch<React.SetStateAction<string>>,
-  setVisualizationType: React.Dispatch<React.SetStateAction<string>>,
-  setVisualizationData: React.Dispatch<React.SetStateAction<Plotly.Data[]>>,
-  setVisualizationMetaData: React.Dispatch<React.SetStateAction<undefined>>,
-  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
-  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>,
-  spanResolution?: string
-) => {
+export const renderCatalogVisualization = async ({
+  http,
+  pplService,
+  catalogSource,
+  startTime,
+  endTime,
+  filterQuery,
+  spanParam,
+  setVisualizationTitle,
+  setVisualizationType,
+  setVisualizationData,
+  setVisualizationMetaData,
+  setIsLoading,
+  setIsError,
+  spanResolution,
+  queryMetaData,
+}: {
+  http: CoreStart['http'];
+  pplService: PPLService;
+  catalogSource: string;
+  startTime: string;
+  endTime: string;
+  filterQuery: string;
+  spanParam: string | undefined;
+  setVisualizationTitle: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationType: React.Dispatch<React.SetStateAction<string>>;
+  setVisualizationData: React.Dispatch<React.SetStateAction<Plotly.Data[]>>;
+  setVisualizationMetaData: React.Dispatch<React.SetStateAction<undefined>>;
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>;
+  setIsError: React.Dispatch<React.SetStateAction<VizContainerError>>;
+  spanResolution?: string;
+  queryMetaData?: MetricType;
+}) => {
   setIsLoading(true);
   setIsError({} as VizContainerError);
 
   const visualizationType = 'line';
   const visualizationTimeField = '@timestamp';
-  let visualizationQuery = `source = ${catalogSource} | stats avg(@value) by span(${visualizationTimeField},1h)`;
 
-  if (spanParam !== undefined) {
-    visualizationQuery = updateQuerySpanInterval(
-      visualizationQuery,
-      visualizationTimeField,
-      spanParam
-    );
-  }
+  const catalogSourceName = catalogSource.split('.')[0];
+  const catalogTableName = catalogSource.split('.')[1];
+
+  const defaultAggregation = 'avg'; // pass in attributes to this function
+  // const attributes: string[] = ['instance', 'job']; // pass in attributes to this function
```

Comment:
we should remove commented code here too

Reply:
this was for reference, ideally it should be deleted